test/concepts.cpp

```diff
@@ -1,14 +1,37 @@
 // Copyright 2022-2023 Alfredo A. Correa
+// Copyright 2024 Matt Borland
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+
+#include <boost/multi/array.hpp>
+
+// Suppress warnings from boost.test
+#if defined(__clang__)
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Wold-style-cast"
+#  pragma clang diagnostic ignored "-Wundef"
+#  pragma clang diagnostic ignored "-Wconversion"
+#  pragma clang diagnostic ignored "-Wsign-conversion"
+#  pragma clang diagnostic ignored "-Wfloat-equal"
+#elif defined(__GNUC__)
+#  pragma GCC diagnostic push
+#  pragma GCC diagnostic ignored "-Wold-style-cast"
+#  pragma GCC diagnostic ignored "-Wundef"
+#  pragma GCC diagnostic ignored "-Wconversion"
+#  pragma GCC diagnostic ignored "-Wsign-conversion"
+#  pragma GCC diagnostic ignored "-Wfloat-equal"
+#endif
+
+#ifndef BOOST_TEST_MODULE
+#  define BOOST_TEST_MAIN
+#endif
 
 #include <boost/test/unit_test.hpp>
-
-#include <boost/mpl/list.hpp>
-
-#include <multi/array.hpp>
+#include <boost/mp11.hpp>
 
 namespace multi = boost::multi;
 
-using NDArrays = boost::mpl::list<
+using NDArrays = boost::mp11::mp_list<
```

Comment:
I highly recommend not using MPL in new code. It's a C++03 library so it doesn't have variadic templates and takes forever to compile. MP11 is way better and has more features: https://www.boost.org/doc/libs/1_85_0/libs/mp11/doc/html/mp11.html#mp_listt